### PR TITLE
SAK-30067 clean-up assignments javascript, velocity macros, remove old spinners and code in favour of new centralized approach

### PR DESCRIPTION
**Pull request by bjones86**
https://jira.sakaiproject.org/browse/SAK-30067

PR introduces significant code refactoring in Assignments to address the following issues:

* moving most JavaScript functions out of .vm files and into the appropriate .js file(s) (with the exception of some that are making use of inline Velocity variables, which I didn't have time to refactor appropriately)
* namespacing all JavaScript functions (in the .js files) to avoid collisions (best practise)
* introduction of a new submitForm() JavaScript function to handle common form submission routines
* removing most inline, long chains of JavaScript functionality in onclicks in favour of calling a common method in a JavaScript file; mostly the newly introduced submitForm function mentioned above (best practise and for readability)
* moving macros defined in individual .vm files into the assignment_macros.vm file
* creation of two new macros to handle the navBar and buttons, to eliminate code duplication in several files
* creation of a new macro to handle the paginator controls which are common between several interfaces
* removing all old spinners and JavaScript code in favour of the new centralized approach introduced in SAK-30059
* replace occurances of href="#" with href="javascript:void(0)" to avoid page bouncing in certain browsers

### CODE REVIEW

File: assignment/assignment-tool/tool/src/webapp/js/assignments.js [hunk 1]
```diff
@@ -1,26 +1,26 @@
 // 'Namespace'
 var ASN = {};
```
Comment (bjones86) [inline assignment/assignment-tool/tool/src/webapp/js/assignments.js]: Cool, haven't seen that before. I'll update.

**Comment by bjones86** [comment]
Updated as per Earle's comment.

**Comment by bjones86** [comment]
If anyone else has time to give this a quick review I'd be very appreciative. Hoping to get this in sooner than later as it touches a lot of files in assignments and merging might get messy if the files diverge too much.

**Comment by bjones86** [comment]
I tried to update this PR today and noticed it has conflicts now. What's the best/easiest way to merge changes from master into a feature branch? I really don't want to have to re-do this entire patch :(

**Comment by ottenhoff** [comment]
Try copying your branch and then 

git rebase upstream/master

**Comment by bjones86** [comment]
Thanks, Sam. I'll try it out.

**Pull request merged**